### ISSUE TITLE
AzCopy on Linux x86_64 platform (circleci hosted CI/CD)

### ISSUE THREAD

**Issue by mpowney** [Azure/azure-storage-azcopy#164]
### Which version of the AzCopy was used? 
Using AzCopy 10.0.4-Preview - azcopy_linux_amd64_10.0.4

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux platform (circleci 2.0 hosted service for CI / CD)

### What command did you run?
```bash
./azcopy sync "/home/circleci/circle-ci-project-name/path-to-files/" "https://storageaccount.blob.core.windows.net/\$web/?sastoken" --recursive
```

### What problem was encountered?
The following error was returned:
```bash
/bin/bash: ./azcopy: cannot execute binary file: Exec format error
Exited with code 126
```

### How can we reproduce the problem in the simplest way?
Seems to be related to the circleci's platform, so I think the simplest way to reproduce would likely be to create a build there with just a single file, and attempt syncing to an azure storage blob.

### Have you found a mitigation/solution?
No mitigation found yet.  Still investigating options.

FWIW below is the gcc -dumpmachine output for this environment:
```bash
circleci@f2927b2fe02d:~$ gcc -dumpmachine
x86_64-linux-gnu
```

And this is the lscpu output for the environment:
```bash
circleci@f2927b2fe02d:~$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                36
On-line CPU(s) list:   0-35
Thread(s) per core:    2
Core(s) per socket:    18
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Platinum 8124M CPU @ 3.00GHz
Stepping:              4
CPU MHz:               3000.000
BogoMIPS:              6000.00
Hypervisor vendor:     KVM
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              25344K
NUMA node0 CPU(s):     0-35
```

**Comment by zezha-msft** [comment]
Hi @mpowney, thanks for reaching out!

Could you please check the kernel version with `uname -a`?

**Comment by mpowney** [comment]
Hey there, yes the output of uname -a is as follows:

```
circleci@dbe374a8eb01:~$ circleci@dbe374a8eb01:~$ uname -a
Linux dbe374a8eb01 4.4.0-141-generic #167~14.04.1-Ubuntu SMP Mon Dec 10 13:20:24 UTC 2018 x86_64 GNU/Linux
```

**Comment by zezha-msft** [comment]
Hi @mpowney, thanks for getting back to me so quickly!

Unfortunately, I don't see anything unusual here, so I'm quite puzzled by why your vm is complaining about the executable format. The distributed binary is for the right CPU architecture and the right OS.

```bash
file azcopy
azcopy: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, not stripped
```

Could you please validate that the executable was not corrupted? For the latest version 10.0.5, the MD5 is:
```bash
md5sum azcopy
4cc5975ccf71fa3f1c5a2770664c144b  azcopy
```

**Comment by mpowney** [comment]
Hi @zezha-msft 

It is now working with v 10.0.5-Preview. The md5 checksum of this version matches yours: 

```
4cc5975ccf71fa3f1c5a2770664c144b  azcopy
```

To prove I'm not going crazy, is it possible to provide a direct link to the v10.0.4-Preview binary, so I can re-confirm the behaviour, and to compare MD5 hashes it?

Regardless of whether the 10.0.4 binary is still available, I'd consider this issue to be resolved now, and have confirmed 10.0.5 is fully functional, so happy to close this out.

**Comment by zezha-msft** [comment]
Hi @mpowney, glad to hear that your issue was resolved! Please don't hesitate if you have any other question.

Here is a [link](https://azcopyvnext.azureedge.net/release20181102/azcopy_linux_amd64_10.0.4.tar.gz) to the 10.0.4-preview executable.
